### PR TITLE
ARM esil32: ldr reg, [pc, reg] correct

### PR DESCRIPTION
**Pull request by pluswave**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

as title.

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — CHANGES_REQUESTED** [review]
Please fix the tests: https://github.com/rizinorg/rizin/runs/7427772183?check_suite_focus=true#step:12:751
Looks like new output is better

File: test/db/esil/arm_32 [hunk 1]
```diff
@@ -589,6 +589,27 @@ r2 = 0xffeeddcc
...
+CMDS=<<EOF
+e asm.arch=arm
+e asm.bits=32
+s 0x20
+wx 00009fe7 # ldr r0, [pc, r0]
+ar r0=0x20
+s 0x48
+wx 01000000
+s 0x28
+wx 02000000
```
Comment (ret2libc) [inline test/db/esil/arm_32]:
```suggestion
wx 01000000 @ 0x48
wx 02000000 @ 0x28
```

File: test/db/esil/arm_32 [hunk 2]
```diff
@@ -589,6 +589,27 @@ r2 = 0xffeeddcc
 EOF
 RUN
 
+NAME=ldr r0, [pc, r0]
+FILE=malloc://0x200
+CMDS=<<EOF
+e asm.arch=arm
+e asm.bits=32
+s 0x20
+wx 00009fe7 # ldr r0, [pc, r0]
```
Comment (ret2libc) [inline test/db/esil/arm_32:598]: could you also add a `pd 1` after this instruction?
I think this is another good test for what you are fixing in the other PR.

Comment (pluswave) [reply test/db/esil/arm_32:598]: added.
````
            0x00000020      ldr   r0, [pc, r0]                         ; [0x28:4]=2
````

The comment part
```` [0x28:4]=2 ````
should not be show , but this is another issue.

**Review by XVilka — APPROVED** [review]

**Pull request merged**